tests/openedx_tagging/core/tagging/test_system_defined.py

```diff
@@ -0,0 +1,236 @@
+    )
+    @ddt.unpack
+    def test_user_object_is_valid(self, value, assert_value):
+        if assert_value:
+            user = get_user_model()(
+                username='username',
+                email='email'
+            )
+            user.save()
+            value = user.id
+
+        object_tag = TestUserObjectTag(
+            taxonomy=self.user_system_taxonomy,
+            object_id='id',
+            object_type='object',
+            value=value,
+        )
+
+        result = object_tag.is_valid(
+            check_taxonomy=True,
+            check_object=True,
+            check_tag=True,
+        )
+
+        self.assertEqual(result, assert_value)
+
+
+@override_settings(LANGUAGES=test_languages)
+class TestLanguageObjectClass(TestCase):
+    """
+    Test for Language object class
+    """
+
+    fixtures = [
+        "tests/openedx_tagging/core/fixtures/system_defined.yaml",
+        "openedx_tagging/core/tagging/fixtures/language_taxonomy.yaml"
+    ]
+
+    def setUp(self):
+        super().setUp()
+        self.taxonomy = Taxonomy.objects.get(name="System Languages")
+        self.expected_langs_ids = sorted([item[0] for item in test_languages])
+        self.expected_langs_values = sorted([item[1] for item in test_languages])
+        self.english_tag = Tag.objects.get(value="English")
+        self.spanish_tag = Tag.objects.get(value="Spanish")
+
+    def test_get_available_languages(self):
+        langs = LanguageObjectTag._get_available_languages()
+        self.assertEqual(sorted(langs), self.expected_langs_ids)
+
+    def test_is_valid(self):
+        valid_object_tag = LanguageObjectTag(
+            taxonomy=self.taxonomy,
+            object_id='id 1',
+            object_type='object',
+            tag=self.english_tag,
+        )
+
+        invalid_onbject_tag_1 = LanguageObjectTag(
+            taxonomy=self.taxonomy,
+            object_id='id 2',
+            object_type='object',
+            tag=self.spanish_tag,
+        )
+
+        invalid_onbject_tag_2 = LanguageObjectTag(
+            taxonomy=self.taxonomy,
+            object_id='id 2',
+            object_type='object',
+        )
+
+        # Tag is not in available languages
+        assert not invalid_onbject_tag_1.is_valid(
+            check_taxonomy=True,
+            check_object=True,
+            check_tag=True,
+        )
+
+        # Testing parent class validations
+        assert not invalid_onbject_tag_2.is_valid(
+            check_taxonomy=True,
+            check_object=True,
+            check_tag=True,
+        )
+
+        # Valid
+        assert valid_object_tag.is_valid(
+            check_taxonomy=True,
+            check_object=True,
+            check_tag=True,
+        )
+
+    def test_get_tags_query_set(self):
+        tags = LanguageObjectTag.get_tags_query_set(self.taxonomy)
+        for tag in tags:
+            self.assertIn(tag.value, self.expected_langs_values)
```

Comment:
Could you also add some tests [like these](https://github.com/openedx/openedx-learning/pull/62/files#diff-6cda6f74e9f6bf053923d015688ea76e50cfec92ef244df48aaa3942288fcbb5R348-R385) to ensure that the `tagging_api.cast_object_tag` method casts the system language tags to your `LanguageObjectTag` class as expected?

Since everything hinges on us using the correct ObjectTag subclass, we need to make sure this remains true.

Reply:
This comment is obsolete now.. closing.